### PR TITLE
[RED-1965] - Add support to a new resource - /api/v2/tags

### PR DESCRIPTION
**Pull request by fbvilela**
### Description

adds support to `Tags`  -> `/api/v2/tags` 

The idea is that this PR will be used as a guide for future implementations of endpoints that support cursor pagination.

### CODE REVIEW

**Comment by fbvilela** [comment]
@mikerogers123 hello again, would appreciate your feedback. this PR depends on https://github.com/justeat/ZendeskApiClient/pull/312

so just take a look at the last commit 🙏

**Comment by fbvilela** [comment]
thanks @mikerogers123 , I have updated the branch. it's all good to review 👍  thanks in advance

**Comment by fbvilela** [comment]
@mikerogers123 gentle bump on this again :)

I have an exciting Pull Request coming right after this! 

 but first it would be good to merge/ get any feedback in this branch! 
Thanks in advance! 🙏

File: src/ZendeskApi.Client/Resources/Interfaces/ITagsResource.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+namespace ZendeskApi.Client.Resources.Interfaces
+{
+    public interface ITagsResource
+    {
+        #region List
+        /// <summary>
+        /// Lists all tags. This request is paginated.
+        /// </summary>
+        Task<TagResponse> GetAllAsync(
+            CursorPager cursor = null,
```
Comment (mikerogers123) [inline src/ZendeskApi.Client/Resources/Interfaces/ITagsResource.cs:15]: i don't think we want to give consumers the option to not pass it in. I don't think a cursor request makes sense without this object supplied

File: src/ZendeskApi.Client/Resources/TagsResource.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+namespace ZendeskApi.Client.Resources
+{
+    public class TagsResource : AbstractBaseResource<TagsResource>, ITagsResource
+    {
+        private static string ResourceUri = "/api/v2/tags";
+        public TagsResource(IZendeskApiClient apiClient, ILogger logger) : base(apiClient, logger, "tags")
+        {
+        }
+
+        public async Task<TagResponse> GetAllAsync(CursorPager cursor = null, CancellationToken cancellationToken = default)
```
Comment (mikerogers123) [inline src/ZendeskApi.Client/Resources/TagsResource.cs:17]: I think the other cursor endpoint return `ICursorPagination<T>`. We should probably do the same here for consistency. See here for another example: https://github.com/justeat/ZendeskApiClient/blob/5d8af4deb12ddae55973985d33112ef68283b48b/src/ZendeskApi.Client/Resources/Ticket/TicketsResource.cs#L46C17-L46C17

**Review by mikerogers123 — CHANGES_REQUESTED** [review]
Looks decent overall but a few things worth addressing :)

File: src/ZendeskApi.Client/Resources/TagsResource.cs (hunk 2 above)
Comment (fbvilela) [reply src/ZendeskApi.Client/Resources/TagsResource.cs]: Thanks! all makes sense! good for another look 🙏

**Comment by mikerogers123** [comment]
@fbvilela looks good to me. Can you bump version? Will need to be to 7.0.4 because there is a competing PR going in at the moment

**Comment by fbvilela** [comment]
All good on this one, @mikerogers123 ?

**Review by mikerogers123 — APPROVED** [review]

**Pull request merged**